synfig-core/src/synfig/string_helper.cpp

```diff
@@ -39,8 +39,16 @@ std::string
 synfig::remove_trailing_zeroes(const std::string& text, bool force_decimal_point)
 {
 	std::string result(text);
+// TODO(ice0): move all the locale related code to the initialization part
+// MinGW C++ std::locale accepts "C" and "POSIX" it does not support other locales.
+#ifdef __MINGW32__
+	struct lconv *locale_info = localeconv();
```

Comment:
Maybe we should work with `ChangeLocale` guard here? To ensure "system" locale instead of "C" locale? We don't know what caller did before calling `synfig::remove_trailing_zeroes()`

Reply:
I don't think this is a good idea to change the locale on every function call. But as a temporary solution, I think it's okay.

Reply:
ice0, I agree it isn't good, but it is only necessary for `__MINGW32__` case.
The "else" case doesn't need it: read line 50 below.

So, please move the `ChangeLocale` for `#if __MINGW32__` case

Reply:
`setlocale(LC_NUMERIC, nullptr)` 
> If locale is NULL, the current locale is only queried, not modified.

Can you explain your idea why `ChangeLocale` is not required for the `#else` branch?

Reply:
Oh well. Forget about ChangeLocale lol I'm confused.